x-pack/plugins/ml/public/embeddables/get_embeddable_component.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import type { CoreStart } from '@kbn/core/public';
+import {
+  EmbeddableFactory,
+  EmbeddableInput,
+  EmbeddableRoot,
+  useEmbeddableFactory,
+} from '@kbn/embeddable-plugin/public';
+import { EuiLoadingChart } from '@elastic/eui';
+import type { MappedEmbeddableTypeOf } from './types';
+import type { MlStartDependencies } from '../plugin';
+import type { MlEmbeddableTypes } from './constants';
+
+/**
+ * Gets an instance of an embeddable component of requested type.
+ * @param embeddableType
+ * @param core
+ * @param plugins
+ */
+export function getEmbeddableComponent<EmbeddableType extends MlEmbeddableTypes>(
+  embeddableType: EmbeddableType,
+  core: CoreStart,
+  plugins: MlStartDependencies
+) {
+  const { embeddable: embeddableStart } = plugins;
+  const factory =
+    embeddableStart.getEmbeddableFactory<MappedEmbeddableTypeOf<EmbeddableType>>(embeddableType);
+
+  if (!factory) {
+    throw new Error(`Embeddable type "${embeddableType}" has not been registered.`);
+  }
+
+  return (props: MappedEmbeddableTypeOf<EmbeddableType>) => {
```

Comment:
Should we memoized the component?

Reply:
wrapped with `React.memo` in https://github.com/elastic/kibana/pull/139628/commits/23bd0df94b51276de4dc5d4e6521d842771fc7b5